In [28]:
from scipy import stats
import numpy as np

import warnings
warnings.filterwarnings('ignore')
warnings.warn('DelfStack')
warnings.warn('Do not show this message')

In [29]:
import matplotlib.pyplot as plt
import seaborn as sns
import pandas as pd

## Тестирование гипотез

In [34]:
df = pd.read_excel('a_b test ecom.xlsx')

In [35]:
df.head(5)

id     time  con_treat      page  converted
0  851104  11:48.6    control  old_page          0
1  804228  01:45.2    control  old_page          0
2  661590  55:06.2  treatment  new_page          0
3  853541  28:03.1  treatment  new_page          0
4  864975  52:26.2    control  old_page          1

In [36]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 294478 entries, 0 to 294477
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype 
---  ------     --------------   ----- 
 0   id         294478 non-null  int64 
 1   time       294478 non-null  object
 2   con_treat  294478 non-null  object
 3   page       294478 non-null  object
 4   converted  294478 non-null  int64 
dtypes: int64(2), object(3)
memory usage: 11.2+ MB


In [37]:
df.describe()

id      converted
count  294478.000000  294478.000000
mean   787974.124733       0.119659
std     91210.823776       0.324563
min    630000.000000       0.000000
25%    709032.250000       0.000000
50%    787933.500000       0.000000
75%    866911.750000       0.000000
max    945999.000000       1.000000

In [38]:
df.isna().sum()

id           0
time         0
con_treat    0
page         0
converted    0
dtype: int64

In [40]:
df.page.value_counts()

old_page    147239
new_page    147239
Name: page, dtype: int64

In [44]:
# Анализ конверсии по группам
conversion_by_group = pd.crosstab(df['con_treat'], df['converted'])
print('Conversion by group:')
print(conversion_by_group)
print()

Conversion by group:
converted       0      1
con_treat               
control    129479  17723
treatment  129762  17514



Исходя из этих данных, для группы control конверсия составляет 12.04%, для группы treatment - 11.86%. Разница между группами незначительна.

In [48]:
from scipy.stats import chi2_contingency

# Создаем кросс-таблицу
contingency_table = pd.crosstab(df['con_treat'], df['converted'])


In [49]:
# Проводим хи-квадрат тест
chi2, p, dof, ex = chi2_contingency(contingency_table)
print(f"Test statistic: {chi2}")
print(f"P-value: {p}")

Test statistic: 1.5159618356336582
P-value: 0.2182316121631168


Значение статистики хи-квадрат равно 1.52, а p-value равен 0.22. Учитывая, что уровень значимости стандартно равен 0.05, мы не можем отвергнуть нулевую гипотезу (что группы не отличаются друг от друга), так как значение p-value больше чем уровень значимости.

Таким образом, мы не можем считать, что различия между группами статистически значимы.